Refactor/cleanup

lilioid opened the pull request:
you might need to adapt cronjobs because the `extract.sh` file is no longer present

lilioid committed:
switch from requirements.txt to Pipfile

lilioid committed:
refactor generate_extracts.py

done so that this script can be used as a standalone, helpful binary

lilioid committed:
use threadpool for parallel work

lilioid committed:
remove extract.sh since it is not compatible anymore

lilioid committed:
add dockerfile

lilioid committed:
remove unused tqdm dependency

Akasch commented on generate_extracts.py:
```diff
@@ -1,55 +1,98 @@
+    subprocess.run(['wget', '-N', '-nv', '--show-progress', url], check=True, cwd=str(working_dir))
+
+    file_name = url.rsplit('/', 1)[-1]
+    subprocess.run(['ln', '-sf', (working_dir / file_name).absolute(), (working_dir / 'source.osm.pbf').absolute()])
```
I think you want to name the file 0_0_0.pbf or you have to change the code below accessing the file 0_0_0.pbf

lilioid replied:
i have done so

Akasch reviewed (changes requested):
Maybe merge it with #1 for the parallel processing as here it will wait on all tiles of a zoom level to have finished, while in #1 the calculation of the children begins immediately if processes are free, speeding up the whole process because otherwise most of the time all Threads are waiting for the one having central Europe in it.

lilioid committed:
improve multiprocessing

now not only each zoom level gets computed in parallel but instead each
tile parallizes itself until the target size or maximum zoom level has
been reached

lilioid commented:
ping @Akasch

lilioid committed:
fix errors found during generation

Akasch reviewed (approved):
If it is tested to be working it looks good
